src/components/TransactionHistory.jsx

```diff
@@ -0,0 +1,369 @@
+import React, { useEffect, useState, useContext } from "react";
+import styled, { css } from "styled-components";
+import { camelToWords } from "../utils";
+import { getWalletInfo } from '../wallets/wallets'
+import { CDPsToList } from "./RepayContent";
+import PrimaryButton from "./PrimaryButton";
+import chevron from '../assets/icons/tablePag_icon.png'
+import { ThemeContext } from "../contexts/ThemeContext";
+import { getCDPs } from "../transactions/cdp";
+import { app } from './Firebase';
+import {
+  getFirestore,
+  getDoc,
+  doc
+} from "firebase/firestore";
+
+// get the firestore database instance
+const db = getFirestore(app);
+
+function mAlgosToAlgos(num) {
+  return num / 1000000
+}
+
+function mAlgosToAlgosFixed(num) {
+  return mAlgosToAlgos(num).toFixed(2)
+}
+
+export async function loadDbActionAndMetrics() {
```

Comment:
Should this be moved to Firebase.jsx? So, that all the DB read and write functions are grouped together?

Reply:
makes sense to me, should I go ahead and move it there ?